src/snowflake/snowpark/context.py

```diff
@@ -18,3 +22,65 @@ def get_active_session() -> "snowflake.snowpark.Session":
         A :class:`Session` object for the current session.
     """
     return snowflake.snowpark.session._get_active_session()
+
+
+class TaskContext:
+    """
+    Represents the context in a `Snowflake Task <https://docs.snowflake.com/en/user-guide/tasks-intro>`_.
+    """
+
+    def __init__(self, session: "snowflake.snowpark.Session") -> None:
+        """
+        Initializes a :class:`TaskContext` object.
+
+        Args:
+            session: A Snowpark session
+        """
+        self._session = session
+
+    @private_preview(version="1.4.0")
+    def set_return_value(self, value: Any) -> None:
+        """
+        Explicitly sets the return value for a task. This value can be retrieved
+        in a child task using :meth:`get_predecessor_return_value`.
+        This method can only be called in a Snowflake task.
+
+        See `SYSTEM$SET_RETURN_VALUE <https://docs.snowflake.com/en/sql-reference/functions/system_set_return_value>`_ for details.
+
+        Args:
+            value: The return value for a task. It will be converted to a ``str``
+                when the underlying SQL system function is called.
+        """
+        self._session.call("system$set_return_value", str(value))
+
+    @private_preview(version="1.4.0")
+    def get_predecessor_return_value(self, task_name: Optional[str] = None) -> str:
+        """
+        Retrieves the return value for the predecessor task in a DAG of tasks.
+        The return value is explicitly set by the predecessor task using :meth:`set_return_value`.
+        This method can only be called in a Snowflake task.
+
+        See `SYSTEM$GET_PREDECESSOR_RETURN_VALUE <https://docs.snowflake.com/en/sql-reference/functions/system_get_predecessor_return_value>`_ for details.
+
+        Args:
+            task_name: The task name of the predecessor task that sets the return value to be retrieved.
+
+                - If the task has only one predecessor task that is enabled, the argument is optional.
+                  If this argument is omitted, the function retrieves the return value for the only enabled predecessor task.
+                - If the task has multiple predecessor tasks that are enabled, this argument is required.
```

Comment:
I'm curious: what is the behavior if no argument is provided and the task has multiple predecessor tasks? Can we add a test to cover this case?

Reply:
It will throw an error. We're not able to test it because it can only be called in task. On the SnowAPI side, we'll have such tests (I have tested locally)